demo/examples/05-pdf.js

```diff
@@ -5,167 +5,67 @@ const root = document.getElementById('example-05');
-  btn.working = true;
+  btn.showSpinner();
+
+  const mapWidth = 277; // mm
+  const mapHeight = 150; // mm
 
-  // Get map size from printSpec
+  // Force map size to fit the PDF document
   const specValue = {
     ...spec.value,
-    size: [pdfSpec.map.width, pdfSpec.map.height],
+    size: [mapWidth, mapHeight, 'mm'],
   };
 
   // create a job, get a promise that resolves when the job is finished
   const blob = await print(specValue);
 
-  // hide the loading spinner
-  btn.working = false;
+  btn.hideSpinner();
 
-  // convert the result into a Data URL
-  const reader = new FileReader();
-  reader.readAsDataURL(blob);
-  reader.onloadend = function () {
-    const imgData = reader.result;
+  // initializes the PDF document
+  const doc = new jsPDF({
+    orientation: 'landscape',
+    unit: 'mm',
+    format: 'a4',
+    putOnlyUsedFonts: true,
+  });
 
-    // add the image to a newly created PDF
-    const doc = new jsPDF({
-      orientation: pdfSpec.orientation,
-      unit: 'px',
-      format: pdfSpec.format,
-      putOnlyUsedFonts: true,
-      hotfixes: ['px_scaling'],
-    });
+  // create an Object URL from the map image blob and add it to the PDF
+  const imgUrl = URL.createObjectURL(blob);
+  doc.addImage(imgUrl, 'JPEG', 10, 40, mapWidth, mapHeight);
 
-    doc.addImage(
-      imgData,
-      'JPEG',
-      pdfSpec.map.x,
-      pdfSpec.map.y,
-      pdfSpec.map.width,
-      pdfSpec.map.height
-    );
+  // add a title
+  doc.setFont('times', 'bold');
+  doc.setFontSize(20);
+  doc.text('A fantastic map.', 148.5, 20, null, null, 'center');
 
-    // add other blocks from the pdf spec
-    for (const block of pdfSpec.blocks) {
-      if (block.rectangle) {
-        doc.rect(
-          block.rectangle.x,
-          block.rectangle.y,
-          block.rectangle.width,
-          block.rectangle.height
-        );
-      }
-      if (block.text) {
-        if (block.text.font) {
-          doc.setFont(block.text.font.name, block.text.font.style);
-          doc.setFontSize(block.text.font.size);
-        }
-        doc.text(
-          block.text.content,
-          block.text.x,
-          block.text.y,
-          null,
-          null,
-          block.text.align
-        );
-        if (pdfSpec.font) {
-          doc.setFont(pdfSpec.font.name, pdfSpec.font.style);
-          doc.setFontSize(pdfSpec.font.size);
-        }
-      }
-    }
+  // add a creation date
+  doc.setFont('courier', 'normal');
+  doc.setFontSize(12);
+  doc.text(
+    `Print date : ${new Date().toLocaleString()}`,
+    10,
+    200,
+    null,
+    null,
+    'left'
+  );
 
-    // download the result
-    const filename = `inkmap-${new Date().toISOString().substr(0, 10)}.pdf`;
-    doc.save(filename);
-  };
-});
+  // add attribution
+  doc.setFont('courier', 'normal');
+  doc.setFontSize(12);
+  doc.text(
+    `Credit to whoever helped make this wonderful map.`,
+    287,
+    200,
+    null,
+    null,
+    'right'
+  );
 
-const pdfSpec = {
```

Comment:
I like the idea of keeping the examples light, but I let @LHBruneton-C2C answer here. Being used in d-inkmap the work is not lost in any case.

Reply:
I agree with the pdf example simplification. People shouldn't be scared of using jsPDF, and we are not here to show all of this library capabilities. It was mainly a playground before d-inkmap.